### ISSUE TITLE
[Bug?] Potential Misalignment in Critic Value Masking: Index Offset in PPO Value Calculation

### ISSUE THREAD

**Issue by tiesanguaixia** [verl-project/verl#1758]
I've identified a potential indexing issue in the critic model's value masking during PPO training. The current implementation slices `attention_mask` with an offset that may misalign with the response tokens:

https://github.com/volcengine/verl/blob/78532923368aeb058f62201489546d013df47710/verl/workers/critic/dp_critic.py#L175

For reference, OpenRLHF's implementation:

https://github.com/OpenRLHF/OpenRLHF/blob/494850f50342ed38d5ae76ef45a3207f3523b582/openrlhf/models/model.py#L280-L285

I have seen another issue about this: [Question about the calculation of critic_loss](https://github.com/volcengine/verl/issues/847)

**Comment by vermouth1992** [comment]
Yes, I think you are right. It should be attention_mask[:, -response_length:]

**Comment by vermouth1992** [comment]
Do you mind submitting a PR to fix this? Thank you very much!

**Comment by tiesanguaixia** [comment]
Thank u, and I may try it recently~
